libs/libc/dirent/lib_nftw.c

```diff
@@ -0,0 +1,230 @@
+/****************************************************************************
+ * libs/libc/dirent/lib_nftw.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <ftw.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <limits.h>
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+static int
+call_nftw(FAR char *path, nftw_cb_t fn, int flags, int base,
+          int level, FAR const struct stat *buf, int info)
+{
+  struct FTW ftw =
+  {
+    base, level
+  };
+
+#ifndef CONFIG_DISABLE_ENVIRON
```

Comment:
i guess it's more user-friendly to make this compile-time (not define FTW_CHDIR) or runtime error (return EINVAL or something explicitly)
i prefer the former.

Reply:
Done.